extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -91,6 +91,29 @@ public static function onSitemapRobotsPageBeforeOutput( WebRequest $request, Use
 		return true;
 	}
 
+	public static function parserUpgradeVignetteUrls( string &$url ) {
+		if ( preg_match( self::VIGNETTE_IMAGES_HTTP_UPGRADABLE, $url ) && strpos( $url, 'http://' ) === 0 ) {
+			$url = wfHttpToHttps( $url );
+		}
+	}
+
+	/**
+	 * Make sure any "external" links to our own wikis that support HTTPS
+	 * are protocol-relative on output.
+	 *
+	 * @param  string  &$url
+	 * @param  string  &$text
+	 * @param  bool    &$link
+	 * @param  array   &$attribs
+	 * @return boolean
+	 */
+	public static function onLinkerMakeExternalLink( string &$url, string &$text, bool &$link, array &$attribs ): bool {
```

Comment:
not used args. text, link, attribs.

Reply:
This is a standard we use since the hook always passes all of these params here: https://github.com/Wikia/app/blob/2fb5243/includes/Linker.php#L1064 so all callers usually specify all the params the hook provides. It's not strictly necessary in cases like this where we're only using the first param (but would be if we were only using the `$attribs` param for example), but it's consistent.

Reply:
Right... Why then this function doesn't have it? https://github.com/Wikia/app/pull/15845/files#diff-fca4065dcdb1fb06adf96c5420beadf6R94

Reply:
That's a different hook call, and due to the way it's checked, it still works as expected: https://github.com/Wikia/app/blob/2fb5243/includes/parser/Parser.php#L1882 However, it probably could explicitly return a value in this case since it's used in a conditional.

But, not returning a value here in `LinkerMakeExternalLink` https://github.com/Wikia/app/blob/2fb5243/includes/Linker.php#L1064 would break the functionality as we don't want to overwrite the output and stop the call to `Html::rawElement`, we just want to change the value of `$url` before the `Html::rawElement` call happens.

Reply:
That's a different hook that only has one parameter: https://github.com/Wikia/app/blob/2fb5243/includes/parser/Parser.php#L1882